### PR TITLE
Update prboom-plus to 2.6.66

### PR DESCRIPTION
**Pull request by Peppersawce**
Latest Haiku version available through HaikuDepot of PrBoom-plus (2.6.1um) fails to find its own wad on hrev58291 x86_64 so I tried making a recipe for the latest available version (2.6.66).

There's a bunch of issues currently with this recipe I'm not sure how to solve:
- The recipe for version 2.6.1um was called "2.6um" which ends up marking this new package as obsolete with its current name in pkgman
- Fixed: The default configuration sets up the preferred midi player as SDL but that one doesn't work and makes the game hang after being closed (fixed by setting the default as opl2, the other midi players don't work atm, but they can work afaik)

Multiplayer is untested.

### CODE REVIEW

File: games-fps/prboom-plus/prboom_plus-2.6.66.recipe [hunk 1]
```diff
@@ -54,12 +60,19 @@ BUILD_PREREQUIRES="
...
+	$prefix/bin/prboom-plus-game-server
+
 BUILD()
 {
 	mkdir -p build && cd build
+
 	cmake ../prboom2 $cmakeDirArgs \
-		-DCMAKE_INSTALL_BIN=$commandBinDir \
-		-DDOOMWADDIR="$dataDir/prboom-plus"
+		-DCMAKE_BUILD_TYPE=Release \
```
[the remark this answers is not in the document]
Comment (jmairboeck) [reply games-fps/prboom-plus/prboom_plus-2.6.66.recipe]: Sorry, I had a typo here, the correct build type is [RelWithDebInfo](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html).

**Comment by Peppersawce** [comment]
Apparently there's an environment variable called DOOMDATAPATH with which one could define multiple paths at compile-time, separated by `:`, but I couldn't get that to work so in order to have more search paths for prboom-plus by default I put some hardcoded paths in to replace some hardcoded "/usr/share" and "/usr/local/share" paths for now.

**Comment by Peppersawce** [comment]
Small note: prboom-plus doesn't show up on HaikuDepot when you search "doom" for some reason. I thought that kind of search looked in both summary and description of the package for it?

Is that a thing I could fix with the new recipe by any chance?

**Comment by Begasus** [comment]
> Small note: prboom-plus doesn't show up on HaikuDepot when you search "doom" for some reason.

That shows up OK here in HaikuDepot and with pkgman.

Anything left to do for this one?

**Comment by Peppersawce** [comment]
> Anything left to do for this one?

Using DOOMWADPATH during compile-time would be neat if somebody makes that work but it's not necessary either (I tried multiple ways of writing that but nothing)
Imo the one thing that should be fixed about this package is the issue with the older package being seen as a newer version. I wouldn't put this one in HaikuDepot until that's sorted.

**Comment by Begasus** [comment]
> Using DOOMWADPATH during compile-time would be neat if somebody can make that work but it's not necessary either (I tried multiple ways of writing that but nothing seemed to be recognized)

Not checked, so no idea on that one so far.



> Imo the one thing that should be fixed about this package is the issue with the older package being seen as a newer version. I wouldn't put this one in HaikuDepot until that's sorted.

Add a REPLACES for it, that will supersede and resolve the conflict for the older version?

**Comment by Peppersawce** [comment]
Will try REPLACES and update the recipe if it works, ty :)

**Comment by jmairboeck** [comment]
I wouldn't bother too much with the nominally lower version number. I'm not sure if this is solvable at all. But for people using `pkgman full-sync` or SoftwareUpdater, it doesn't really matter anyway. Just `pkgman update` wouldn't work in this case.

**Comment by Begasus** [comment]
OK, checked, both versions can't be installed at the same time, so no use for REPLACES, thanks!

**Pull request merged**